### PR TITLE
Add exceptions, cleanup code and negative tests

### PR DESCRIPTION
**Pull request by Inno97**
Added proper exceptions and cleanup for the following classes:

	commons
		commands
			RouteAddCommand.java
			RouteDeleteCommand.java
			RouteEditCommand.java
			RouteListAllCommand.java
			RouteListCommand.java
			RouteNodeAddCommand.java
			RouteNodeDeleteCommand.java
			RouteNodeEditCommand.java
			RouteNodeListCommand.java
		Messages.java
		
	logic
		api
			requests
				LocationSearchUrlRequest
				StaticMapRequest
				UrlRequest

	model
		lists
			RouteList.java
	
		locations
			BusStop.java
			RouteNode.java
			TrainStation.java
			Venue.java
		
		transports
			BusService.java
			Route.java
			
	storage
		Storage.java

Add negative tests for RouteCommand classes.
		
Needs fixing:
RecommendationsCommand
	 - create a more specific exception for "itinerary cannot be more than 8 days"

### CODE REVIEW

File: src/main/java/duke/commons/Messages.java [hunk 1]
```diff
@@ -4,29 +4,134 @@
...
+
+    public static final String PROMPT_REPEAT_STARTER = "What task that repeats every X days would you like to add?";
+    public static final String PROMPT_REPEAT_DATE = "When does the task start?";
+    public static final String PROMPT_REPEAT_REPEAT = "How many days does it repeat?";
+    public static final String PROMPT_REPEAT_SUCCESS = "Repeating task added!";
+
+    public static final String PROMPT_DELETE_STARTER = "What would you like to delete?";
+    public static final String PROMPT_DELETESUCCESS = "Trying to delete task...";
+
+    public static final String FINDPATH_PROMPT_STARTER = "Where do you want to go?";
```
Comment (Jefferson111) [inline src/main/java/duke/commons/Messages.java]: odd one out

File: src/main/java/duke/logic/CreateMap.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ private void setTrainLine() {
 
         for (Map.Entry mapElement : this.trainMap.entrySet()) {
             TrainStation trainStation = (TrainStation)mapElement.getValue();
-            for (String trainCode : trainStation.getTrainCode()) {
+            for (String trainCode : trainStation.getTrainCodes()) {
```
Comment (Jefferson111) [inline src/main/java/duke/logic/CreateMap.java]: Good!

File: src/main/java/duke/logic/api/ApiParser.java [hunk 3]
```diff
@@ -22,37 +23,43 @@
...
-            LocationSearchUrlRequest req = new LocationSearchUrlRequest(param);
-            JsonObject jsonRes = req.execute();
-            JsonArray arr = jsonRes.getAsJsonArray("results");
+    public static Venue getLocationSearch(String param) throws ApiNullRequestException, ApiTimeoutException {
+        LocationSearchUrlRequest req = new LocationSearchUrlRequest(param);
+        JsonObject jsonRes = req.execute();
+        JsonArray arr = jsonRes.getAsJsonArray("results");
 
-            assert (isFound(jsonRes) && arr.size() != 0);
+        if (isValidLocationSearch(jsonRes) && arr.size() != 0) {
```
Comment (Jefferson111) [inline src/main/java/duke/logic/api/ApiParser.java]: LOD

Comment (Inno97) [reply src/main/java/duke/logic/api/ApiParser.java]: Refactored.

**Pull request merged**